build.gradle

```diff
@@ -114,51 +114,56 @@ artifacts {
-    repositories {
-        mavenDeployer {
-            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
-
-            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
-                authentication(userName: ossrhUsername, password: ossrhPassword)
-            }
-
-            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
-                authentication(userName: ossrhUsername, password: ossrhPassword)
-            }
-
-            pom.project {
-                name 'Studio for kdb+'
-                packaging 'jar'
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            from components.java
+            pom {
+                name = 'Studio for kdb+'
+                packaging = 'jar'
                 // optionally artifactId can be defined here
-                description 'Studio for kdb+ is a rapid development environment for the ultra-fast database kdb+ from Kx Systems: http://www.kx.com.'
-                url 'https://github.com/dzmipt/kdbStudio'
+                description = 'Studio for kdb+ is a rapid development environment for the ultra-fast database kdb+ from Kx Systems: http://www.kx.com.'
+                url = 'https://github.com/dzmipt/kdbStudio'
 
                 scm {
-                    connection 'git@github.com:dzmipt/kdbStudio.git'
-                    developerConnection 'git@github.com:dzmipt/kdbStudio.git'
-                    url 'https://github.com/dzmipt/kdbStudio'
+                    connection = 'git@github.com:dzmipt/kdbStudio.git'
+                    developerConnection = 'git@github.com:dzmipt/kdbStudio.git'
+                    url = 'https://github.com/dzmipt/kdbStudio'
                 }
 
                 licenses {
                     license {
-                        name 'The Apache License, Version 2.0'
-                        url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                        name = 'The Apache License, Version 2.0'
+                        url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
                     }
                 }
 
                 developers {
                     developer {
-                        id 'dzmipt'
-                        name 'Dmitry Zakharov'
-                        email 'dz.mipt@gmail.com'
+                        id = 'dzmipt'
+                        name = 'Dmitry Zakharov'
+                        email = 'dz.mipt@gmail.com'
                     }
                 }
             }
         }
     }
+    repositories {
+        maven {
+            url "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+            mavenContent {
+                releasesOnly()
+            }
+        }
+        maven {
+            url "https://oss.sonatype.org/content/repositories/snapshots/"
+            mavenContent {
+                snapshotsOnly()
+            }
+        }
+    }
```

Comment:
you loose authentication details... Check in the old version there was
`authentication(userName: ossrhUsername, password: ossrhPassword)`
I think they should be somewhere here

Reply:
Will look at it. Are these properties required to be set (even if you are not going to publish anything) in $HOME/.gradle/gradle.properties?